src/snowflake/connector/cursor.py

```diff
@@ -607,6 +607,66 @@ def _preprocess_pyformat_query(
             query = command
         return query
 
+    @overload
+    def execute(
+        self,
+        command: str,
+        params: Sequence[Any] | dict[Any, Any] | None = None,
+        _bind_stage: str | None = None,
+        timeout: int | None = None,
+        _exec_async: bool = False,
+        _no_retry: bool = False,
+        _do_reset: bool = True,
+        _put_callback: SnowflakeProgressPercentage = None,
+        _put_azure_callback: SnowflakeProgressPercentage = None,
+        _put_callback_output_stream: IO[str] = sys.stdout,
+        _get_callback: SnowflakeProgressPercentage = None,
+        _get_azure_callback: SnowflakeProgressPercentage = None,
+        _get_callback_output_stream: IO[str] = sys.stdout,
+        _show_progress_bar: bool = True,
+        _statement_params: dict[str, str] | None = None,
+        _is_internal: bool = False,
+        _describe_only: bool = False,
+        _no_results: Literal[False] = False,
```

Comment:
Why use `Literal[False]` instead of `bool`?

Reply:
I think probably because the value of `_no_results` determines the actual return type of this function?

Reply:
shixuan hit the nail, it overrides the signature to make the type hint clear on the return type depending on the `_no_results`